testapp/src/main/java/com/rakuten/tech/mobile/testapp/ui/settings/Cache.kt

```diff
@@ -170,92 +149,174 @@ internal class Cache(context: Context) {
         private const val CONTACTS = "contacts"
         private const val TOKEN_DATA = "token_data"
         private const val URL_PARAMETERS = "url_parameters"
-        private const val ANALYTIC_CONFIGS = "analytic_configs"
         private const val ACCESS_TOKEN_ERROR = "access_token_error"
         private const val POINTS = "points"
         private const val DYNAMIC_DEEPLINKS = "dynamic_deeplinks"
         private const val MAX_STORAGE_SIZE_LIMIT = "max_storage_size_limit"
+        private const val IS_TEMP_CLEARED = "is_temp_cleared"
     }
 
-    inner class RasCredentialData(context: Context) {
+    @Suppress("TooManyFunctions")
+    inner class RasCredentialData(context: Context, manifestConfig: AppManifestConfig) {
+        val isDefaultProductionEnabled = BuildConfig.BUILD_TYPE == BuildVariant.RELEASE.value
 
+        private val tab1MiniAppConfigCache = MiniAppConfigCache(
+            IS_PROD_VERSION_ENABLED,
+            IS_PREVIEW_MODE,
+            REQUIRE_SIGNATURE_VERIFICATION,
+            APP_ID,
+            SUBSCRIPTION_KEY
+        )
 
-        val defaultProdPair = Pair(
-            context.getString(R.string.prodProjectId),
-            context.getString(R.string.prodSubscriptionKey)
+        private val tab1TempMiniAppConfigCache = MiniAppConfigCache(
+            TEMP_IS_PROD_VERSION_ENABLED,
+            TEMP_IS_PREVIEW_MODE,
+            TEMP_REQUIRE_SIGNATURE_VERIFICATION,
+            TEMP_APP_ID,
+            TEMP_SUBSCRIPTION_KEY
```

Comment:
Just a question regarding this is this an android specific implementation choice to use extra properties for the temporary config? couldn't this be implemented in a way such as `var listConfigI = MiniAppConfigCache(...)` to make it mutable? I feel this is bit of an unnecessary overhead.

Reply:
1. There needs a data to make it temporary when the user havent clicked the save button yet.
2. There needs a data to make it persistent when user switched to another tab and comeback. This one.
3. Should the save button functionality removed,  It wouldnt need these, agree.
4. When save button is clicked. It would clear these temporaries (both 1 and 2)

Reply:
ok but you could use a getter that uses a fallback to temporary data (from prefs) and otherwise use the actual data right? just as an example in iOS `Config.string(listType, key: .projectId, fallbackKey: .projectId) ?? ""` which basically falls back to the fallBackKey in case no persisted data exists

Reply:
As far as I know in Android the fallback should directly assign the primitive nullable value, except if it calls another key and acts as a fallback, which would be in a call loop since it ends to call primitive value as default. Also, this can also have the same concept as the fallback key...

Besides, the layout here is using the same view components for both tabs. each one view holds different states (previous saved data, modified unsaved data, modified saved data).

Reply:
gotcha seems platform related